multigeiger/transmission.cpp

```diff
@@ -70,59 +71,89 @@ void send_http(HTTPClient *http, String body) {
   http->end();
 }
 
-void send_http_geiger(const char *host, String tube_type, unsigned int timediff, unsigned int hv_pulses, unsigned int gm_counts, unsigned int cpm) {
-  bool addname = false;
+void send_http_geiger(const char *host, String tube_type, unsigned int timediff, unsigned int hv_pulses,
+                      unsigned int gm_counts, unsigned int cpm, bool addname, int xpin) {
   char body[1000];
   HTTPClient http;
-  prepare_http(&http, host);
-  http.addHeader("X-PIN", "19");
+  prepare_http(&http, host, xpin);
   tube_type = tube_type.substring(10);
-  String prefix = (addname ? tube_type + "_" : "");
+  String prefix = addname ? tube_type + "_" : "";
   const char *json_format = R"=====(
 {
  "software_version": "%s",
  "sensordatavalues": [
   {"value_type": "%scounts_per_minute", "value": "%d"},
-  {"value_type": "hv_pulses", "value": "%d"},
-  {"value_type": "counts", "value": "%d"},
-  {"value_type": "sample_time_ms", "value": "%d"}
+  {"value_type": "%shv_pulses", "value": "%d"},
+  {"value_type": "%scounts", "value": "%d"},
+  {"value_type": "%ssample_time_ms", "value": "%d"}
  ]
 }
 )=====";
   snprintf(body, 1000, json_format,
            http_software_version.c_str(),
            prefix.c_str(), cpm,
-           hv_pulses,
-           gm_counts,
-           timediff);
+           prefix.c_str(), hv_pulses,
+           prefix.c_str(), gm_counts,
+           prefix.c_str(), timediff);
   send_http(&http, body);
 }
 
-void send_http_thp(const char *host, float temperature, float humidity, float pressure) {
-  bool addname = false;
-  String prefix = (addname ? "BME280_" : "");
+void send_http_thp(const char *host, float temperature, float humidity, float pressure, int xpin) {
   char body[1000];
   HTTPClient http;
-  prepare_http(&http, host);
-  http.addHeader("X-PIN", "11");
+  prepare_http(&http, host, xpin);
   const char *json_format = R"=====(
 {
  "software_version": "%s",
  "sensordatavalues": [
-  {"value_type": "%stemperature", "value": "%.2f"},
-  {"value_type": "%shumidity", "value": "%.2f"},
-  {"value_type": "%spressure", "value": "%.2f"}
+  {"value_type": "temperature", "value": "%.2f"},
+  {"value_type": "humidity", "value": "%.2f"},
+  {"value_type": "pressure", "value": "%.2f"}
  ]
 }
 )=====";
   snprintf(body, 1000, json_format,
            http_software_version.c_str(),
-           prefix.c_str(), temperature,
-           prefix.c_str(), humidity,
-           prefix.c_str(), pressure);
+           temperature,
+           humidity,
+           pressure);
   send_http(&http, body);
 }
 
+void send_http_geiger_and_thp(const char *host, String tube_type, unsigned int timediff, 
+  unsigned int hv_pulses, unsigned int gm_counts, unsigned int cpm,
+  float temperature, float humidity, float pressure) {
+  char body[1000];
+  HTTPClient http;
+  prepare_http(&http, host, 0);
```

Comment:
pin 0?

Reply:
MADAVI doesn't interpret the X-PIN header so the number can be any

Reply:
but then the header rather should not be there.

Reply:
Yeah, would be better.